src/core/qgsfieldmodel.h

```diff
@@ -135,7 +135,7 @@ class CORE_EXPORT QgsFieldModel : public QAbstractItemModel
      * like the field name, alias and type.
      * \since QGIS 3.0
      */
-    static QString fieldToolTip( const QgsField &field );
+    static QString fieldToolTip( const QgsField &field, const QString &expression = QString() );
```

Comment:
I don't think the expression handling belongs here -- I think it's cleaner to just add this line in QgsAttributeTableModel/QgsIdentifyResultsDialog manually

Reply:
For sure it's a dirty solution, I will remove it. Having a helper for this in the future might be handy, so the tooltips remain the same style across all widgets.

Reply:
I agree about the helper. What about moving it to a static method here?

   static QString appendExpressionToTooltip( const QString& tooltip, const QString& expression )

?

Reply:
Why pass the tooltip string at all? This gives too much freedom which may bring another round of unconsistancies? I would say keep the same approach as I did originally, just adding a new function for that.

```static QString fieldToolTipWithExpression( const QgsField &field, const QString &expression )```

This way everyone is forced to use tooltips that begin with the same contents, then they are free to append their stuff.

Reply:
Hmm, what about

    static QString expandedFieldToolTip( const QgsField &field, QgsVectorLayer* layer )

and just put ALL the logic in there?

Reply:
I like that! But are we sure we will always have the layer in advance? If we are about to create a new layer, it might be missing, so _**maybe**_ overloading will be needed in the future with second parameter being a string?

I only have comment on the name, I would prefer `fieldToolTipExpanded` over `expandedFieldToolTip`, it looks better from API points of view.

Reply:
Have a look now, I have renamed it to `fieldToolTipExtended` as "expanded" seems strange to me? You are native speaker, you know better if the new name makes sense :)